File: flo-device-service/src/action_rule.go
package main

import "time"

type ActionRule struct {
	Id             string    `json:"id,omitempty"`
	TargetDeviceId string    `json:"targetDeviceId"`
	Action         string    `json:"action"`
	Event          string    `json:"event"`
	Order          int       `json:"order"`
	Enabled        bool      `json:"enabled"`
	CreatedAt      time.Time `json:"createdAt"`
	UpdatedAt      time.Time `json:"updatedAt"`
}

type ActionRules struct {
	Data []ActionRule `json:"actionRules"`
}

File: flo-puck-service/src/main/scala/com/flo/puck/core/api/ActionRule.scala
package com.flo.puck.core.api

sealed trait Action
case object ShutOff         extends Action
case object DiscardedAction extends Action

sealed trait Event
case object WaterDetected  extends Event
case object DiscardedEvent extends Event

case class ActionRule (
  id: String,
  targetDeviceId: String,
  action: Action,
  event: Event,
  order: Int,
  enabled: Boolean
)